### ISSUE TITLE
Copilot telemetry is sent even when Zed telemetry is disabled

### ISSUE THREAD

**Issue by unikitty37** [zed-industries/zed#21737]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. Install Zed on macOS along with Little Snitch.
2. Configure Little Snitch to prompt before allowing connections.
3. Launch Zed.
4. When Little Snitch prompts for connection permissions for Zed, select Until Quit, All Connections, and press Deny.
5. Turn telemetry and diagnostics off in Zed.
6. Quit Zed.
7. Relaunch Zed

Expected behaviour: no telemetry connections are made.
Actual behaviour: Zed requests permission to connect to copilot-telemetry.githubusercontent.com.

If Copilot does not respect Zed's telemetry being disabled, it shouldn't be used.

### Environment

Zed: v0.164.2 (Zed)
OS: macOS 14.5.0
Memory: 64 GiB
Architecture: aarch64

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

```
2024-12-09T08:55:02.270036Z [INFO] ========== starting zed ==========
2024-12-09T08:55:02.325115Z [INFO] Opening main db
2024-12-09T08:55:02.32627Z [INFO] Opening main db
2024-12-09T08:55:02.36665Z [INFO] Using git binary path: Some("/Applications/Zed.app/Contents/MacOS/git")
2024-12-09T08:55:02.644961Z [INFO] No prompt template overrides directory found at /Users/user/.config/zed/prompt_overrides. Using built-in prompts.
2024-12-09T08:55:02.741354Z [INFO] Opening main db
2024-12-09T08:55:02.744768Z [INFO] installing extension html latest version
2024-12-09T08:55:02.751168Z [ERROR] auto-update failed: error:error sending request for url (https://zed.dev/api/releases/latest?asset=zed&os=macos&arch=aarch64)

Caused by:
    0: client error (Connect)
    1: tcp connect error: Bad file descriptor (os error 9)
    2: Bad file descriptor (os error 9)
2024-12-09T08:55:02.782413Z [ERROR] error downloading extension: error sending request for url (https://api.zed.dev/extensions/html/download?min_schema_version=0&max_schema_version=1&min_wasm_api_version=0.0.1&max_wasm_api_version=0.2.0)
2024-12-09T08:55:02.786125Z [INFO] Opening main db
2024-12-09T08:55:02.80553Z [INFO] Opening main db
2024-12-09T08:55:04.564114Z [INFO] set environment variables from shell:/bin/zsh, path:/Users/user/bin:/opt/homebrew/bin:/Users/user/.local/share/mise/installs/node/22.4.1/bin:/Users/user/.local/share/mise/installs/ruby/3.3.4/bin:/Users/user/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/user/Library/Application Support/JetBrains/Toolbox/scripts:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Library/Apple/usr/bin:/Applications/Little Snitch.app/Contents/Components:/Applications/VMware Fusion Tech Preview.app/Contents/Public:/Users/user/Library/Application Support/JetBrains/Toolbox/scripts:/Users/user/.orbstack/bin
2024-12-09T08:55:09.470041Z [INFO] starting language server process. binary path: "/opt/homebrew/bin/node", working directory: "/", args: ["/Users/user/Library/Application Support/Zed/copilot/copilot-v0.7.0/dist/language-server.js", "--stdio"]
2024-12-09T08:55:10.05336Z [INFO] Language server with id 0 sent unhandled notification window/logMessage:
{
  "type": 3,
  "message": "[lsp] GitHub Copilot Language Server 1.236.0 initialized"
}
2024-12-09T08:55:10.053623Z [INFO] Language server with id 0 sent unhandled notification workspace/configuration:
{
  "items": [
    {
      "section": "github.copilot"
    },
    {
      "section": "github-enterprise"
    },
    {
      "section": "http"
    },
    {
      "section": "telemetry"
    }
  ]
}
2024-12-09T08:55:10.18349Z [INFO] Language server with id 0 sent unhandled notification window/logMessage:
{
  "type": 3,
  "message": "[certificates] Removed 14 expired certificates"
}

```
</details>

**Comment by JosephTLyons** [comment]
It is true, our telemetry setting only the sending of events we've collected - it does not do anything for any third party services used to power features. Not sure how we should handle that, as those are running separately. Wonder if there are settings to configure this, but knowing Microsoft, probably not.

**Comment by unikitty37** [comment]
The comment in the default settings file for `features.inline_completion_provider` just says "Which inline completion provider to use".

It would be useful if it could list the valid options, along with whether those options send their own telemetry (though that requires you to _know_ whether or not they do, of course).

Ideally, this choice would default to "none", and the onboarding process would invite the user to select another option (with that telemetry info clearly given) so they've given explicit opt-in consent before they send any data.

A user who's disabled telemetry in settings and isn't running something like Little Snitch would be completely unaware that telemetry was still being sent; if Microsoft's telemetry contains anything that's classed as personal data under GDPR, that's definitely a potential problem.

**Comment by dandv** [comment]
> A user who's disabled telemetry in settings and isn't running something like Little Snitch would be completely unaware that telemetry was still being sent;

Exactly. I had no idea until I ran OpenSnitch, which revealed connections made by the Copilot plugin, which apparently [can't be disabled](https://github.com/zed-industries/zed/issues/26106)?

**Comment by notpeter** [comment]
Zed has migrated from copilot.vim to copilot-language-server.  Previously, I was not able to successfully prevent telemetry in copilot.vim, but copilot-language-server supports the common microsoft lsp setting to signal telemetry intent:
```json
  "settings": {
    "telemetry": {
      "telemetryLevel": "off"
    }
  }
```
Where telemetry level is one of [`all`, `error`, `crash`, `off`] (Defaults to all)

<img width="794" alt="Image" src="https://github.com/user-attachments/assets/ae4c6e88-f9f1-47d2-a923-66abad5a7ab7" />

Zed could explicitly specify this as a default lsp setting to allow users to control telemetry for all the Microsoft LSPs (json, css, eslint, copilot, etc).  I believe other language servers (e.g. [redhat-developer/yaml-language-server](https://github.com/redhat-developer/vscode-yaml/blob/main/USAGE_DATA.md)) may also respect this signaling, but it's certainly on a case-by-case basis.

**Comment by dandv** [comment]
> Zed could explicitly specify this as a default lsp setting to allow users to control telemetry

Until this is implemented, is there a way to prevent sending telemetry (before the request reaches the firewall)?

> node is connecting to copilot-telemetry.githubusercontent.com

**Comment by SyntheticBird45** [comment]
> > Zed could explicitly specify this as a default lsp setting to allow users to control telemetry
> 
> Until this is implemented, is there a way to prevent sending telemetry (before the request reaches the firewall)?
> 
> > node is connecting to copilot-telemetry.githubusercontent.com

an override in /etc/hosts (or similar on windows) should do the trick as of now.
